### ISSUE TITLE
Support common dumb-init/tini in shebang

### ISSUE THREAD

**Issue by oliv3r** [koalaman/shellcheck#2349]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC1008
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/usr/bin/dumb-init /bin/sh
exit 0
```

#### Here's what shellcheck currently says:
```
^-- SC1008: This shebang was unrecognized. ShellCheck only supports sh/bash/dash/ksh. Add a 'shell' directive to specify.
```

#### Here's what I wanted or expected to see:
Shellcheck doesn't have to know and understand every wrapper in the world, SC1008 rightfully states to just add a `shell=` line. But for docker init scripts, where `#!/.*bin/tini (.*)` and/or  `#!/.*bin/dumb-init (.*)` can be used to help identify the shell quite easily are quote common I'd say, common enough, like the `env` wrapper to make an exception and understand these as well?
